iceoryx2/tests/node_tests.rs

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#[generic_tests::define]
+mod node {
+    use std::collections::HashSet;
+
+    use iceoryx2::config::Config;
+    use iceoryx2::node::{NodeState, NodeView};
+    use iceoryx2::prelude::*;
+    use iceoryx2::service::Service;
+    use iceoryx2_bb_posix::directory::Directory;
+    use iceoryx2_bb_posix::unique_system_id::UniqueSystemId;
+    use iceoryx2_bb_system_types::path::*;
+    use iceoryx2_bb_testing::assert_that;
+
+    #[derive(Debug, Eq, PartialEq)]
+    struct Details {
+        name: NodeName,
+        id: u128,
+        config: Config,
+    }
+
+    impl Details {
+        fn new(name: &NodeName, id: &UniqueSystemId, config: &Config) -> Self {
+            Self {
+                name: name.clone(),
+                id: id.value(),
+                config: config.clone(),
+            }
+        }
+
+        fn from_node<S: Service>(node: &Node<S>) -> Self {
+            Self::new(node.name(), node.id(), node.config())
+        }
+    }
+
+    fn assert_node_presence<S: Service>(node_details: &Vec<Details>, config: Option<&Config>) {
+        let node_list = if let Some(ref config) = config {
+            Node::<S>::list_with_custom_config(config).unwrap()
+        } else {
+            Node::<S>::list().unwrap()
+        };
+
+        assert_that!(node_list, len node_details.len());
+        for node in node_list {
+            match node {
+                NodeState::<S>::Alive(view) => {
```

Comment:
Not sure if this works
```suggestion
                NodeState::<S>::Alive(view) |  => {
```

Reply:
Nope, since those views are two different types, one view for the living and one for the dead.

Reply:
@elfenpiff I added a comment which would make it work. Just for the next time, no need to change it now.